### PR TITLE
Update style book headings to new design

### PR DESCRIPTION
**Pull request by tellthemachines**
## What?

Part of #53431.

Updates the styles of stylebook headings to match designs on that issue.

## Testing Instructions

1. With a block theme active, go to "Styles" in the site editor and select "Style book" from the Preview dropdown.
2. Check that style book headings look as expected.

The one thing in the design that I'm not sure how to best implement is the color difference between headings and actual theme text color:

Screenshot 2024-12-04 at 11 02 01 am

Hardcoding a color in here won't work because the style book uses whatever background color is defined by the theme (and text color as well). The design seems to use always the same color, but what happens if the theme background color happens to be that exact color, or one very close to it?

We should also ensure that these headings have accessible levels of contrast, which some of the variations in the design don't seem to have.

Before I jump into making complicated JS calculations around text/background color in order to dynamically find the appropriate heading color, I just want to make sure we're all on the same page about this 😅 so:

* Is the intention that the heading color be always a little bit lighter than text color when the background is light, and a little darker when the background is dark? 
* Do we want it to always be grey, or should it match the theme text color?

Current headings match theme text color, and I haven't changed that yet in this PR:

Screenshot 2024-12-04 at 10 58 17 am

Screenshot 2024-12-04 at 10 58 28 am

Screenshot 2024-12-04 at 10 58 50 am

### CODE REVIEW

**Comment by jasmussen** [comment]
Thanks for this! I pushed some color changes so that now the headings match the mockups, while meeting contrast:

Screenshot 2024-12-04 at 10 39 58

Screenshot 2024-12-04 at 10 40 02

Doesn't have to be this PR, but can you help me follow up with a few changes to color swatches and the heading example? These two in the mockup:

Screenshot 2024-12-04 at 10 42 50

The color swatches should be shown in 4 columns, and we need a spacer or taller margin between the H1 heading and the two paragraphs in columns, so the spacing gets closer to the mockup. It can be a fixed value if need be. For the moment, they're just next to each other:
Screenshot 2024-12-04 at 10 39 58

**Comment by jasmussen** [comment]
To clarify: on the Style _landing_ page, the colors should be shown in 4 columns:

Screenshot 2024-12-04 at 11 17 05

On the Colors section, they are two columns, and larger:

Screenshot 2024-12-04 at 11 16 59

**Review by matiasbenedetto — APPROVED** [review]
On my testing, the headings of the stylebook sections looked like the mockups, and the colors seemed to be clearly visible with both light and dark background colors. I think this PR is good to go.

https://github.com/user-attachments/assets/72542ba9-44cb-471f-ab26-5a27a8b8e991

> on the Style landing page, the colors should be shown in 4 columns:

We can work on that in another PR because this one seems targeted to headings. I'll submit a PR [for that](https://github.com/WordPress/gutenberg/issues/67592) here: https://github.com/WordPress/gutenberg/pull/67597

File: packages/edit-site/src/components/style-book/constants.ts [hunk 1]
```diff
@@ -216,27 +216,25 @@ export const STYLE_BOOK_IFRAME_STYLES = `
 	}
 	.edit-site-style-book__duotone-example > div:not(:first-child) {
 		height: 20px;
-		border: 1px solid #ddd;
+		border: 1px solid color-mix( in srgb, currentColor 10%, transparent );
```
Comment (tellthemachines) [inline packages/edit-site/src/components/style-book/constants.ts:219]: Oooh I didn't know we were already using `color-mix`! That does make things easier.

**Pull request merged**